crates/wallet/src/wallet/mod.rs

```diff
@@ -2319,9 +2322,10 @@ impl Wallet {
     /// This is the first step when performing a spk-based wallet partial sync, the returned
     /// [`SyncRequest`] collects all revealed script pubkeys from the wallet keychain needed to
     /// start a blockchain sync with a spk based blockchain client.
-    pub fn start_sync_with_revealed_spks(&self) -> SyncRequest {
-        SyncRequest::from_chain_tip(self.chain.tip())
-            .populate_with_revealed_spks(&self.indexed_graph.index, ..)
+    pub fn build_sync_with_revealed_spks(&self) -> SyncRequestBuilder<(KeychainKind, u32)> {
+        SyncRequest::builder()
+            .chain_tip(self.chain.tip())
+            .revealed_spks_from_indexer(&self.indexed_graph.index, ..)
```

Comment:
I like the new builder pattern for sync and full scan requests, but do we really need to return the builder instead of the final built request (also for full scan) here in Wallet? My understanding is that we'd add new Wallet methods for other useful request options (ie. build_sync_with_revealed_unused_spks(), or something like that).
```suggestion
            .revealed_spks_from_indexer(&self.indexed_graph.index, ..)
            .build()
```

Reply:
I think it's better to return the builder, since the caller can further customize it. I've also changed chain sources to take in `impl Into<...Request>` so they can directly take in a builder struct.